op-challenger/game/fault/solver/game_solver_test.go

```diff
@@ -84,6 +85,32 @@ func TestCalculateNextActions(t *testing.T) {
 				lastHonestClaim.Attack(common.Hash{0xdd}).ExpectStepAttack()
 			},
 		},
+
+		{
+			name:                "PoisonedPreState",
+			agreeWithOutputRoot: true,
+			setupGame: func(builder *faulttest.GameBuilder) {
+				// Create a cannon state that is exited and resolved that the output root was invalid
+				maliciousState := mipsevm.State{
+					Memory:   mipsevm.NewMemory(),
+					ExitCode: 1,
+					Exited:   true,
+				}
+				maliciousStateHash, _ := maliciousState.EncodeWitness().StateHash()
+
+				// Dishonest actor counters their own claims to set up a situation with an invalid prestate
+				// The honest actor should attack all claims that support the root claim (disagree with the output root)
+				builder.Seq().ExpectAttack(). // This expected action is the winning move.
+								Attack(maliciousStateHash).
+								Defend(maliciousStateHash).ExpectAttack().
+								Attack(maliciousStateHash).
+								Attack(maliciousStateHash).ExpectStepAttack()
+
+				// The attempt to step against our malicious leaf node will fail because the pre-state won't match our
```

Comment:
No, the setup I'm going for is where the honest challenger can't call step because it would have to use the attacker's pre-state and that won't be what it tries to provide.

The claims the honest actor evaluates are:
```
    game_solver_test.go:123: Claim 0: Pos: 1 TraceIdx: 15 ParentIdx: 0, Countered: false, Value: 0x000000000000000000000000000000000000000000000000000000000000000f
    game_solver_test.go:123: Claim 1: Pos: 2 TraceIdx: 7 ParentIdx: 0, Countered: false, Value: 0x018bd1331cba4019004e3d2a2cb9fcb748d97fdfe2972b8b856b71d55e06c2b3
    game_solver_test.go:123: Claim 2: Pos: 6 TraceIdx: 11 ParentIdx: 1, Countered: false, Value: 0x018bd1331cba4019004e3d2a2cb9fcb748d97fdfe2972b8b856b71d55e06c2b3
    game_solver_test.go:123: Claim 3: Pos: 12 TraceIdx: 9 ParentIdx: 2, Countered: false, Value: 0x018bd1331cba4019004e3d2a2cb9fcb748d97fdfe2972b8b856b71d55e06c2b3
    game_solver_test.go:123: Claim 4: Pos: 24 TraceIdx: 8 ParentIdx: 3, Countered: false, Value: 0x018bd1331cba4019004e3d2a2cb9fcb748d97fdfe2972b8b856b71d55e06c2b3
```

So trace index 7, 8 and 9 have all been provided by the attacker and the honest actor won't be able to provide a state that matches them.

Actually I probably should have just used random hash values that there is no pre-image for rather than creating a malicious state - if the honest actor was really smart they could actually apply a step based on that malicious state. I'll push that change up.

Reply:
What I meant is that the dishonest actor calls step (as `ExpectStepAttack` asserts) and succeeds in countering a claim supported by the honest challenger.
The motivation here is to DoS/coerce the honest challenger into supporting untruthful claims. If the poisoned claim remains uncontested after a step, then there's no point to creating poisoned claims.

Reply:
I think we agree here, but perhaps I'm interpreting a failed step, as worded here, to mean the parent being disputed remains uncontested.

Reply:
The `Expect*()` calls are setting out what we expect the honest challenger to do - or at least try to do. So it's actually the honest challenger that tries to call step but it will supply the wrong pre-state and so the call will fail.  With the original malicious state that had a pre-image the dishonest actor could have called step successfully to counter their own leaf claim, though that would have then been showing their original root claim to be invalid which they aren't trying to do.

I'm definitely keen to work out how to make all of this easier to understand - the aim of these tests is to give us something more concrete to discuss but that's not helpful if it's still ambiguous.

Reply:
Understood. The poison scenario I originally had in mind is where the honest actor moves last, so it's the dishonest's _turn_ to step. But this setup is analogous and illustrates poison similarly.